Fix -Wimplicit-int errors

The build fails on modern macOS with:

| error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]

This is likely because the default compiler now makes -Wimplicit-int an error
by default. Fix this by adding the return type specifier.
--- tcpd.c.orig	2024-03-25 23:39:11
+++ tcpd.c	2024-03-25 23:39:36
@@ -42,7 +42,7 @@
 int     allow_severity = SEVERITY;	/* run-time adjustable */
 int     deny_severity = LOG_WARNING;	/* ditto */
 
-main(argc, argv)
+int main(argc, argv)
 int     argc;
 char  **argv;
 {
--- fakelog.c	2003-06-18 21:29:18
+++ fakelog.c	2024-03-25 23:46:48
@@ -17,7 +17,7 @@
 
 /* ARGSUSED */
 
-openlog(name, logopt, facility)
+void openlog(name, logopt, facility)
 char   *name;
 int     logopt;
 int     facility;
@@ -27,7 +27,7 @@
 
 /* vsyslog - format one record */
 
-vsyslog(severity, fmt, ap)
+void vsyslog(severity, fmt, ap)
 int     severity;
 char   *fmt;
 va_list ap;
@@ -43,7 +43,7 @@
 
 /* VARARGS */
 
-VARARGS(syslog, int, severity)
+void VARARGS(syslog, int, severity)
 {
     va_list ap;
     char   *fmt;
@@ -56,7 +56,7 @@
 
 /* closelog - dummy */
 
-closelog()
+void closelog()
 {
     /* void */
 }
--- try-from.c	2003-06-18 21:29:19
+++ try-from.c	2024-03-25 23:47:13
@@ -37,7 +37,7 @@
 int     allow_severity = SEVERITY;	/* run-time adjustable */
 int     deny_severity = LOG_WARNING;	/* ditto */
 
-main(argc, argv)
+int main(argc, argv)
 int     argc;
 char  **argv;
 {
--- safe_finger.c	2009-10-05 23:17:07
+++ safe_finger.c	2024-03-25 23:47:34
@@ -54,7 +54,7 @@
     exit(0);
 }
 
-main(argc, argv)
+int main(argc, argv)
 int     argc;
 char  **argv;
 {
